### PR TITLE
Add OnConfigure hook and foreign key support

### PR DESCRIPTION
**Pull request by aforcier**
* Exposed a hook to the `SQLiteOpenHelper.onConfigure()` callback in `DefaultWellConfig`
* In the example app, used `onConfigure()` to enable `FOREIGN KEY` support, and added a test

cc @maxme

### CODE REVIEW

File: well-sample/src/main/java/com/yarolegovich/wellsample/AntiHero.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import com.yarolegovich.wellsql.core.Identifiable;
+import com.yarolegovich.wellsql.core.annotation.Column;
+import com.yarolegovich.wellsql.core.annotation.PrimaryKey;
+import com.yarolegovich.wellsql.core.annotation.RawConstraints;
+import com.yarolegovich.wellsql.core.annotation.Table;
+import com.yarolegovich.wellsql.core.annotation.Unique;
+
+@Table
+@RawConstraints({"FOREIGN KEY(HERO_ID) REFERENCES SuperHero(_id)", "FOREIGN KEY(VILLAIN_ID) REFERENCES Villain(_id)"})
+public class AntiHero implements Identifiable {
```
Comment (maxme) [inline well-sample/src/main/java/com/yarolegovich/wellsample/AntiHero.java]: Should we rename that class to `StrangeCouple` :trollface: (and drop the name column).

Comment (aforcier) [reply well-sample/src/main/java/com/yarolegovich/wellsample/AntiHero.java]: 🤦‍♂️ updated in 8c6eb51!

**Comment by aforcier** [comment]
Ready for another go @maxme 😄

**Comment by maxme** [comment]
:shipit:

**Pull request merged**